services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/model/CrtFileUpload.java

```diff
@@ -57,7 +61,14 @@ public ResumableFileUpload pause() {
 
     private ResumableFileUpload doPause() {
         File sourceFile = request.source().toFile();
-        if (completionFuture.isDone()) {
+
+        boolean futureCompletedExceptionally = completionFuture.isCompletedExceptionally();
```

Comment:
Can we write mock test to test the same ?

Reply:
added test